graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java

```diff
@@ -160,8 +217,9 @@ public void visitMethod(MethodElement element, VisitorContext context) {
     @Override
     public void visitConstructor(ConstructorElement element, VisitorContext context) {
         if (!isSubclass) {
-            if (element.hasAnnotation(Creator.class)) {
+            if (element.hasAnnotation(Creator.class) && element.isPrivate()) {
```

Comment:
@ilopmar again I am not sure why we were creating reflective data is the creator didn't require reflection

Reply:
I don't remember, probably it was necessary sometime ago and maybe it's not now.